### PR TITLE
STROM-1263: port backtype.storm.command.kill-topology to java also STORM-1260, STORM-1261, and STORM-1264

### PR DESCRIPTION
**Pull request by revans2**
(And add in better java CLI)

### CODE REVIEW

File: storm-core/src/jvm/org/apache/storm/command/CLI.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+ * limitations under the License.
+ */
+package org.apache.storm.command;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import org.apache.commons.cli.*;
```
Comment (redsanket) [inline storm-core/src/jvm/org/apache/storm/command/CLI.java]: single imports

**Comment by vesense** [comment]
LGTM

**Comment by vesense** [comment]
Since the PR include STORM-1260 STORM-1261 STORM-1263 STORM-1264, I suggest changing the title of this PR.

File: storm-core/src/jvm/org/apache/storm/command/CLI.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CLI {
+    private static final Logger LOG = LoggerFactory.getLogger(CLI.class);
+    private static class Opt {
+        final String s;
```
Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/command/CLI.java]: can we rename s and l to shortName, longName respectively or something of the sort? or may be let's just put up a comment here.

File: bin/storm.py [hunk 3]
```diff
@@ -278,7 +278,7 @@ def kill(*args):
         print_usage(command="kill")
         sys.exit(2)
     exec_storm_class(
-        "org.apache.storm.command.kill_topology",
+        "org.apache.storm.command.KillTopology",
```
Comment (abhishekagarwal87) [inline bin/storm.py]: you have also added wait as alternative argument name. should it be added in the command doc?

Comment (abhishekagarwal87) [inline bin/storm.py]: My bad. --wait was already there but not documented in command syntax.

**Comment by revans2** [comment]
The travis failure is unrelated.  It is a storm-kafa failure.

**Comment by harshach** [comment]
+1

**Pull request merged**